com.unity.render-pipelines.universal/Shaders/Unlit.shader

```diff
@@ -62,9 +65,9 @@ Shader "Universal Render Pipeline/Unlit"
 
             struct Varyings
             {
-                float2 uv        : TEXCOORD0;
-                float fogCoord  : TEXCOORD1;
-                float4 vertex : SV_POSITION;
+                float4 vertex  : SV_POSITION;
+                float2 uv      : TEXCOORD0;
+                float fogCoord : TEXCOORD1;
```

Comment:
In this case both fogCoord and uv are floats. Shouldn't we pack them in same interpolator? Or does the compiler do this automagically?

Reply:
These should definitely get packed by the compiler for the load, but I suspect these would end up in separate barycentric ops, as they would be in separate slots as different varyings. There are other compiler optimizations where all bary ops can be optimized out of the frag shader but that one is a bit more convoluted. I don't remember there being a packing opt for the bary ops on adreno, but other platforms may do this implicitly for free as long as they are aligned properly as Erik has done. Since everything is a scalar architecture these days, not sure how much benefit we would see, but I still think it is an interesting idea and worth trying (as it may reduce reg pressure). Definitely worth a look at the generated IR, but this might be an opt you only see in the machine code.

Reply:
So apparently, on GCN, the reduction in export is where we will see the benefit if we are export bound (and not ALU bound). This will be less true for RDNA, as it is less likely to be export bound, since exports are done in 4 dwords in batches of 4 (color and position have their own separate paths so not included in this on RDNA). Though as I mentioned, we won't see a reduction in ALU ops for the bary interpolations, but exports if pressured will cost more than an ALU op. There may be some implicit packing of exports, but this is less clear (though I will have a solid response on this at some point). While in the unlit shader, this will matter less, but in general, I think Felipe's suggestion is a really good general practice where possible.

Reply:
FXC implies that those are merged, uv and fogCoord get the same register numbers. i.e. v1.xy = uv and v1.z = fogCoord.
It looks like at least on my machine those are appear to be packed on lower level too.